### ISSUE TITLE
Blank screen when cloning a printer object

### ISSUE THREAD

**Issue by eduardomozart** [glpi-project/glpi#8108]
**Describe the bug**

After cloning a printer object into GLPI, the screen turns blank into https://glpi.example.com/front/massiveaction.php and only a clonned object is created (eg., if I set to create 4 clones, it only creates one).

**Page(s) URL**

https://glpi.example.com/front/printer.php

**To reproduce**

1. Go to 'Assets' > 'Printers'
2. Mark the checkbox of a printer
2. Click on 'Actions' button
3. Select 'Clone' action
4. Set any integer value at 'How many copies do you want to create ?'
5. Click on Submit
6. A blank screen appears into https://glpi.example.com/front/massiveaction.php and, when going back to Assets > Printers page, only one copy is created, even if you specified a number of copies higher than 1.

**Expected behavior**

It should go back to https://glpi.example.com/front/printer.php screen and show up the X number of cloned printer objects.

**Logs**

php-errors.log

```
[2021-04-07 07:47:20] glpiphplog.CRITICAL:   *** Uncaught Exception RuntimeException: Cannot guess field for itemtype Printer on Cartridge in /var/www/clients/client0/web1/web/inc/commondbchild.class.php at line 882
  Backtrace :
  inc/features/clonable.class.php:68                 CommonDBChild::getItemField()
  inc/commondbtm.class.php:1242                      Printer->post_clone()
  inc/massiveaction.class.php:1319                   CommonDBTM->clone()
  inc/massiveaction.class.php:1097                   MassiveAction::processMassiveActionsForOneItemtype()
  inc/massiveaction.class.php:1076                   MassiveAction->processForSeveralItemtypes()
  front/massiveaction.php:59                         MassiveAction->process()

```

sql-errors.log

```
[2021-04-07 07:47:12] glpisqllog.ERROR: DBmysql::query() in /var/www/clients/client0/web1/web/inc/dbmysql.class.php line 309
  *** MySQL query error:
  SQL: SELECT `id` FROM `glpi_plugin_printercounters_items_recordmodels` WHERE `LOWER(itemtype)` = 'LOWER('Printer')' AND `items_id` = '1'
  Error: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'Printer')' AND `items_id` = '1'' at line 1
  Backtrace :
  inc/dbmysqliterator.class.php:95
  inc/dbmysql.class.php:876                          DBmysqlIterator->execute()
  inc/commondbtm.class.php:343                       DBmysql->request()
  plugins/printercounters/ajax/infocom.php:42        CommonDBTM->getFromDBByCrit()
  {"user":"168@ispconfig.example.com"}
[2021-04-07 07:47:14] glpisqllog.ERROR: DBmysql::query() in /var/www/clients/client0/web1/web/inc/dbmysql.class.php line 309
  *** MySQL query error:
  SQL: SELECT `id` FROM `glpi_plugin_printercounters_items_recordmodels` WHERE `LOWER(itemtype)` = 'LOWER('Printer')' AND `items_id` = '1'
  Error: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'Printer')' AND `items_id` = '1'' at line 1
  Backtrace :
  inc/dbmysqliterator.class.php:95
  inc/dbmysql.class.php:876                          DBmysqlIterator->execute()
  inc/commondbtm.class.php:343                       DBmysql->request()
  plugins/printercounters/ajax/infocom.php:42        CommonDBTM->getFromDBByCrit()
  {"user":"168@ispconfig.example.com"}

```

**Screenshots**

The blank screen that appears at https://glpi.example.com/front/massiveaction.php

![2020-10-14 12_29_02-GLPI - Bulk Modification](https://user-images.githubusercontent.com/2974895/96011208-040f1280-0e19-11eb-837a-049e8c24a052.png)

**Your GLPI setup (you can find it in Setup > General menu, System tab)**

```
[code]
 
GLPI 9.5.4 ( => /var/www/clients/client0/web1/web)
Installation mode: TARBALL

Server
 
Operating system: Linux ispconfig.example.com 4.18.0-240.10.1.el8_3.x86_64 #1 SMP Mon Jan 18 17:05:51 UTC 2021 x86_64
PHP 7.3.27 fpm-fcgi (Core, PDO, PDO_ODBC, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apcu, bcmath, bz2, calendar, cgi-fcgi,
	ctype, curl, date, dom, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, imap, intl, json, ldap, libxml, mbstring, mcrypt,
	mysql, mysqli, mysqlnd, odbc, openssl, pcre, pdo_mysql, pdo_sqlite, posix, session, shmop, snmp, soap, sockets, sodium, sqlite3,
	standard, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xml, xmlreader, xmlrpc, xmlwriter, xsl, zip, zlib)
Setup: max_execution_time="600" memory_limit="128M" post_max_size="150 M" safe_mode="" session.save_handler="files"
	upload_max_filesize="150M" 
Software: Apache ()
	Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.82 Safari/537.36
Server Software: MariaDB Server
	Server Version: 10.3.27-MariaDB
	Server SQL Mode: 
	Parameters: c0glpiuser@localhost/c0glpi
	Host info: Localhost via UNIX socket
	
PHP version is at least 7.2.0 - Perfect!PHP version is at least 7.2.0 - Perfect!
Sessions support is available - Perfect!Sessions support is available - Perfect!
Allocated memory > 64 Mio - Perfect!Allocated memory > 64 Mio - Perfect!
mysqli extension is installedmysqli extension is installed
ctype extension is installedctype extension is installed
fileinfo extension is installedfileinfo extension is installed
json extension is installedjson extension is installed
mbstring extension is installedmbstring extension is installed
iconv extension is installediconv extension is installed
zlib extension is installedzlib extension is installed
curl extension is installedcurl extension is installed
gd extension is installedgd extension is installed
simplexml extension is installedsimplexml extension is installed
intl extension is installedintl extension is installed
ldap extension is installedldap extension is installed
apcu extension is installedapcu extension is installed
Zend OPcache extension is installedZend OPcache extension is installed
xmlrpc extension is installedxmlrpc extension is installed
CAS extension is installedCAS extension is installed
exif extension is installedexif extension is installed
zip extension is installedzip extension is installed
bz2 extension is installedbz2 extension is installed
sodium extension is installedsodium extension is installed
Database version seems correct (10.3.27) - Perfect!Database version seems correct (10.3.27) - Perfect!
Timezones seems loaded in databaseTimezones seems loaded in database
The log file has been created successfully.The log file has been created successfully.
Write access to /var/www/clients/client0/web1/web/files/_cache has been validated.Write access to /var/www/clients/client0/web1/web/files/_cache has been validated.
Write access to /var/www/clients/client0/web1/web/config has been validated.Write access to /var/www/clients/client0/web1/web/config has been validated.
Write access to /var/www/clients/client0/web1/web/files/_cron has been validated.Write access to /var/www/clients/client0/web1/web/files/_cron has been validated.
Write access to /var/www/clients/client0/web1/web/files has been validated.Write access to /var/www/clients/client0/web1/web/files has been validated.
Write access to /var/www/clients/client0/web1/web/files/_dumps has been validated.Write access to /var/www/clients/client0/web1/web/files/_dumps has been validated.
Write access to /var/www/clients/client0/web1/web/files/_graphs has been validated.Write access to /var/www/clients/client0/web1/web/files/_graphs has been validated.
Write access to /var/www/clients/client0/web1/web/files/_lock has been validated.Write access to /var/www/clients/client0/web1/web/files/_lock has been validated.
Write access to /var/www/clients/client0/web1/web/files/_pictures has been validated.Write access to /var/www/clients/client0/web1/web/files/_pictures has been validated.
Write access to /var/www/clients/client0/web1/web/files/_plugins has been validated.Write access to /var/www/clients/client0/web1/web/files/_plugins has been validated.
Write access to /var/www/clients/client0/web1/web/files/_rss has been validated.Write access to /var/www/clients/client0/web1/web/files/_rss has been validated.
Write access to /var/www/clients/client0/web1/web/files/_sessions has been validated.Write access to /var/www/clients/client0/web1/web/files/_sessions has been validated.
Write access to /var/www/clients/client0/web1/web/files/_tmp has been validated.Write access to /var/www/clients/client0/web1/web/files/_tmp has been validated.
Write access to /var/www/clients/client0/web1/web/files/_uploads has been validated.Write access to /var/www/clients/client0/web1/web/files/_uploads has been validated.
Write access to /var/www/clients/client0/web1/web/marketplace has been validated.Write access to /var/www/clients/client0/web1/web/marketplace has been validated.
Web access to files directory is protectedWeb access to files directory is protected
For security reasons, SELinux mode should be Enforcing.For security reasons, SELinux mode should be Enforcing.

GLPI constants
 
GLPI_ROOT: /var/www/clients/client0/web1/web
GLPI_CONFIG_DIR: /var/www/clients/client0/web1/web/config
GLPI_VAR_DIR: /var/www/clients/client0/web1/web/files
GLPI_MARKETPLACE_DIR: /var/www/clients/client0/web1/web/marketplace
GLPI_USE_CSRF_CHECK: 1
GLPI_CSRF_EXPIRES: 7200
GLPI_CSRF_MAX_TOKENS: 100
GLPI_USE_IDOR_CHECK: 1
GLPI_IDOR_EXPIRES: 7200
GLPI_ALLOW_IFRAME_IN_RICH_TEXT: 
GLPI_TELEMETRY_URI: https://telemetry.glpi-project.org
GLPI_INSTALL_MODE: TARBALL
GLPI_NETWORK_MAIL: glpi@teclib.com
GLPI_NETWORK_SERVICES: https://services.glpi-network.com
GLPI_MARKETPLACE_PRERELEASES: 
GLPI_USER_AGENT_EXTRA_COMMENTS: 
GLPI_AJAX_DASHBOARD: 1
GLPI_CALDAV_IMPORT_STATE: 0
GLPI_DEMO_MODE: 0
GLPI_FORCE_EMPTY_SQL_MODE: 1
GLPI_DOC_DIR: /var/www/clients/client0/web1/web/files
GLPI_CACHE_DIR: /var/www/clients/client0/web1/web/files/_cache
GLPI_CRON_DIR: /var/www/clients/client0/web1/web/files/_cron
GLPI_DUMP_DIR: /var/www/clients/client0/web1/web/files/_dumps
GLPI_GRAPH_DIR: /var/www/clients/client0/web1/web/files/_graphs
GLPI_LOCAL_I18N_DIR: /var/www/clients/client0/web1/web/files/_locales
GLPI_LOCK_DIR: /var/www/clients/client0/web1/web/files/_lock
GLPI_LOG_DIR: /var/www/clients/client0/web1/web/files/_log
GLPI_PICTURE_DIR: /var/www/clients/client0/web1/web/files/_pictures
GLPI_PLUGIN_DOC_DIR: /var/www/clients/client0/web1/web/files/_plugins
GLPI_RSS_DIR: /var/www/clients/client0/web1/web/files/_rss
GLPI_SESSION_DIR: /var/www/clients/client0/web1/web/files/_sessions
GLPI_TMP_DIR: /var/www/clients/client0/web1/web/files/_tmp
GLPI_UPLOAD_DIR: /var/www/clients/client0/web1/web/files/_uploads
GLPI_NETWORK_REGISTRATION_API_URL: https://services.glpi-network.com/api/registration/
GLPI_MARKETPLACE_PLUGINS_API_URI: https://services.glpi-network.com/api/glpi-plugins/
GLPI_I18N_DIR: /var/www/clients/client0/web1/web/locales
GLPI_VERSION: 9.5.4
GLPI_SCHEMA_VERSION: 9.5.4
GLPI_MIN_PHP: 7.2.0
GLPI_YEAR: 2021

Libraries
 
htmlawed/htmlawed version 1.2.5 in (/var/www/clients/client0/web1/web/vendor/htmlawed/htmlawed)
phpmailer/phpmailer version 6.1.6 in (/var/www/clients/client0/web1/web/vendor/phpmailer/phpmailer/src)
simplepie/simplepie version 1.5.6 in (/var/www/clients/client0/web1/web/vendor/simplepie/simplepie/library)
tecnickcom/tcpdf version 6.3.5 in (/var/www/clients/client0/web1/web/vendor/tecnickcom/tcpdf)
michelf/php-markdown in (/var/www/clients/client0/web1/web/vendor/michelf/php-markdown/Michelf)
true/punycode in (/var/www/clients/client0/web1/web/vendor/true/punycode/src)
iamcal/lib_autolink in (/var/www/clients/client0/web1/web/vendor/iamcal/lib_autolink)
sabre/dav in (/var/www/clients/client0/web1/web/vendor/sabre/dav/lib/DAV)
sabre/http in (/var/www/clients/client0/web1/web/vendor/sabre/http/lib)
sabre/uri in (/var/www/clients/client0/web1/web/vendor/sabre/uri/lib)
sabre/vobject in (/var/www/clients/client0/web1/web/vendor/sabre/vobject/lib)
laminas/laminas-cache in (/var/www/clients/client0/web1/web/vendor/laminas/laminas-cache/src)
laminas/laminas-i18n in (/var/www/clients/client0/web1/web/vendor/laminas/laminas-i18n/src)
laminas/laminas-serializer in (/var/www/clients/client0/web1/web/vendor/laminas/laminas-serializer/src)
monolog/monolog in (/var/www/clients/client0/web1/web/vendor/monolog/monolog/src/Monolog)
sebastian/diff in (/var/www/clients/client0/web1/web/vendor/sebastian/diff/src)
elvanto/litemoji in (/var/www/clients/client0/web1/web/vendor/elvanto/litemoji/src)
symfony/console in (/var/www/clients/client0/web1/web/vendor/symfony/console)
scssphp/scssphp in (/var/www/clients/client0/web1/web/vendor/scssphp/scssphp/src)
laminas/laminas-mail in (/var/www/clients/client0/web1/web/vendor/laminas/laminas-mail/src/Protocol)
laminas/laminas-mime in (/var/www/clients/client0/web1/web/vendor/laminas/laminas-mime/src)
rlanvin/php-rrule in (/var/www/clients/client0/web1/web/vendor/rlanvin/php-rrule/src)
blueimp/jquery-file-upload in (/var/www/clients/client0/web1/web/vendor/blueimp/jquery-file-upload/server/php)
ramsey/uuid in (/var/www/clients/client0/web1/web/vendor/ramsey/uuid/src)
psr/log in (/var/www/clients/client0/web1/web/vendor/psr/log/Psr/Log)
psr/simple-cache in (/var/www/clients/client0/web1/web/vendor/psr/simple-cache/src)
mexitek/phpcolors in (/var/www/clients/client0/web1/web/vendor/mexitek/phpcolors/src/Mexitek/PHPColors)
guzzlehttp/guzzle in (/var/www/clients/client0/web1/web/vendor/guzzlehttp/guzzle/src)
guzzlehttp/psr7 in (/var/www/clients/client0/web1/web/vendor/guzzlehttp/psr7/src)
wapmorgan/unified-archive in (/var/www/clients/client0/web1/web/vendor/wapmorgan/unified-archive/src)
paragonie/sodium_compat in (/var/www/clients/client0/web1/web/vendor/paragonie/sodium_compat/src)
phpCas version 1.3.8 in (/usr/share/pear)

LDAP directories
 
Server: 'ldaps://example.com', Port: '389', BaseDN: 'OU=Example,DC=example,DC=com', Connection filter:
		'(&(objectClass=user)(objectCategory=person)(!(userAccountControl:1.2.840.113556.1.4.803:=2))(!(memberOf=CN=Contas de Serviço -
		Negar Logon Interativo,OU=Serviços,OU=TI,OU=São
		Leopoldo,OU=Example,DC=example,DC=com,DC=br))(!(sAMAccountName=fabrica))(!(sAMAccountName=cq))(!(sAMAccountName=manutencao)))',
		RootDN: 'glpiaduser@example.com', Use TLS: none

SQL replicas
 
Not active

Notifications
 
Way of sending emails: SMTP+SSL (glpi@example.com@smtp.kinghost.net)

Mails receivers
 
Name: 'glpi@example.com' Active: Yes
	Server: '{pop.kinghost.net:995/pop/ssl/validate-cert}' Login: 'glpi@example.com' Password: Yes

Plugins list
 
	fields               Name: Campos adicionais              Version: 1.12.3     State: Enabled
	behaviors            Name: Comportamentos                 Version: 2.4.1      State: Enabled
	accounts             Name: Contas                         Version: 2.6.0      State: Enabled
	archimap             Name: Diagrams                       Version: 2.2.1      State: Enabled
	fusioninventory      Name: FusionInventory                Version: 9.5+2.0    State: Enabled
	genericobject        Name: Gerenciamento de objetos       Version: 2.10.1     State: Enabled
	manufacturersimports Name: Importação de fornecedores     Version: 2.3.1      State: Enabled
	pdf                  Name: Imprimir em PDF                Version: 1.7.0      State: Enabled
	mreporting           Name: Mais Relatórios                Version: 1.7.2      State: Enabled
	printercounters      Name: Printer counters               Version: 1.7.0      State: Enabled
	reports              Name: Relatórios                     Version: 1.14.1     State: Enabled
	taskdrop             Name: TaskDrop                       Version: 1.2.0      State: Enabled

[/code]
```

**Additional context**

I applied the proposed fix from #8218 but the issue still persists.

**Comment by G-Sheridan** [comment]
Should this issue be labeled as Good First Issue?

We are doing research about how to help newcomers onboarding in an open source project and how to accelerate the issue resolving process.
We have developed a tool to automatically label the Good First Issue (GFI), i.e., the issue which is good for newcomers. 
We first extract 72 features from different dimensions of issue reports, such as readability, semantic meaning, associated code snippets, etc, and experimentally investigate to what extent these features correlate with GFI. Results turn out that a large portion of these features do have a medium to large correlation with GFI. 
We then build a machine learning model based on these features to predict GFI. The experiments on a dataset of 12k issues from Github show that 0.90 AUC can be achieved. 

Our tool shows that this issue has a very large possibility to be a GFI. 
Could you please tell us whether it should be labeled as a GFI?
And if you have any comments or suggestions with this tool, could you please tell us? 
Thank you so much.

**Comment by Kaya84** [comment]
I will confirm this happen to me too.
```
[2021-04-06 11:30:21] glpiphplog.CRITICAL:   *** Uncaught Exception RuntimeException: Cannot guess field for itemtype Printer on Cartridge in /var/www/glpi/inc/commondbchild.class.php at line 882
  Backtrace :
  inc/features/clonable.class.php:68                 CommonDBChild::getItemField()
  inc/commondbtm.class.php:1234                      Printer->post_clone()
  inc/massiveaction.class.php:1306                   CommonDBTM->clone()
  inc/massiveaction.class.php:1084                   MassiveAction::processMassiveActionsForOneItemtype()
  inc/massiveaction.class.php:1063                   MassiveAction->processForSeveralItemtypes()
  front/massiveaction.php:59                         MassiveAction->process()
```

Clone will work, but management data are not copied

**Comment by eduardomozart** [comment]
Hello @Kaya84,
Into my case, it's copying the Management data, but the blank screen still occours when cloning one or more printer items.
I updated the ticket description from GLPI 9.5.2 to GLPI 9.5.4, since this bug still persists into newer GLPI versions. But now it's throwing SQL errors into the log, something that wasn't happening before, so maybe it could help to trace the issue.

**Comment by cedric-anne** [comment]
Fixed by #9558 .
